extra/end-to-end-test.rb

```diff
@@ -0,0 +1,66 @@
+require 'rest-client'
+require 'json'
+require 'pp'
+
+# subnets and domains must exist on Infoblox
+URL = "http://localhost:8448"
+DNS_DOMAIN = "ibx.lan"
+DNS_IPV4 = "192.168.202.13"
+DNS_IPV6 = "2001:db8::1"
+DNS_PTR4 = "13.202.168.192.in-addr.arpa"
+DNS_PTR6 = "1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa"
+
+def do_get(path)
+  RestClient.get("#{URL}#{path}")
+end
+
+def do_post(path, body = nil, params = {})
+  RestClient.post("#{URL}#{path}", body, params.merge({content_type: :json}))
+end
+
+def do_delete(path, params = {})
+  RestClient.delete("#{URL}#{path}", params.merge({content_type: :json}))
+end
+
+features = JSON.parse(do_get("/v2/features"))
+
+if features["dns"]["state"] == "running"
+  fqdn = "end2end.#{DNS_DOMAIN}"
+  cname = "alias-end2end.#{DNS_DOMAIN}"
+  puts "Performing DNS tests with #{fqdn}"
+  begin
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=#{DNS_IPV4}")
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=#{DNS_IPV4}") # skip (already exists)
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=8.8.8.8") rescue RestClient::Conflict # expected
```

Comment:
Perhaps it should raise an error if there is no conflict? This will silently accept HTTP 200

I'd also not use any real IPs. rfc5735 suggests using TEST-NET-1 (`192.0.2.0/24`), TEST-NET-2 (`198.51.100.0/24`) or TEST-NET-3 (`203.0.113.0/24`) for this.

Reply:
That's how our common DNS code was designed, I won't change it from here. If you look to `dns_common.rb` it silently returns (`nil`). Probably something our orchestration code in Foreman core expects as well.

I replaced Google DNS with `192.0.2.1`.